src/components/ForexRatesTab.js

```diff
@@ -39,22 +43,86 @@ const styles = (theme) => ({
 });
 
 function ForexRatesTab(props) {
-  const { classes } = props;
+  const { classes, getRates } = props;
+
+  const [forexRates, setForexRates] = useState([]);
+  const [snackBarParams, setSnackBarParams] = useState({
+    show: false, message: '', variant: 'success',
+  });
+
+  useEffect(() => {
+    async function fetchForexRates() {
+      try {
+        const fxpResponse = await getRates();
+        setForexRates(fxpResponseToForexRates(fxpResponse));
+      } catch (error) {
+        if (error.name === 'AbortError') {
+          setSnackBarParams({
+            show: true,
+            message: 'Timeout getting Forex rates. Retry?',
+            variant: 'error',
+            callback: fetchForexRates,
+            action: 'retry',
+          });
+        } else {
+          setSnackBarParams({
+            show: true,
+            message: 'An error occurred trying to get the Forex rates. Retry?',
+            variant: 'error',
+            callback: fetchForexRates,
+            action: 'retry',
+          });
+        }
+      }
+    }
+    fetchForexRates();
+  }, []);
+
+  // Forex Rates get transformed into this:
+  // [
+  //   {
+  //     currencyPair: 'eurusd',
+  //     rate: '666.6667',
+  //     startTime: '2019-09-03T12:00:00.000Z',
+  //     endTime: '2019-09-04T12:00:00.000Z',
+  //     reuse: true,
+  //   },
+  //   {
+  //     currencyPair: 'eurusd',
+  //     rate: '666.6680',
+  //     startTime: '2019-09-04T12:00:00.000Z',
+  //     endTime: '2019-09-05T12:00:00.000Z',
+  //     reuse: true,
+  //   },
+  //   {
+  //     currencyPair: 'usdeur',
+  //     rate: '444.4430',
+  //     startTime: '2019-09-03T12:00:00.000Z',
+  //     endTime: '2019-09-04T12:00:00.000Z',
+  //     reuse: true,
+  //   },
+  // ]
+
   return (
-    <div className={classes.root}>
-      <p>Forex Rates Tab</p>
-    </div>
+    <Grid className={classes.root} container spacing={0}>
+      <Grid item xs={12}>
+        <ForexRateEntry />
+      </Grid>
+      <Grid item xs={12}>
+        <ForexRatesTable forexRates={forexRates} />
+      </Grid>
+      {console.log(snackBarParams)}
```

Comment:
Did you intend for this to remain?

Reply:
If so, consider adding a little more context to what you're printing. So it's a little more obvious to a reader, and so it's easier to find in the source (yes, I know there are sourcemaps etc., but that does slow things down a bit, and IIRC the suggestion here will still display nicely in the console).
```suggestion
      {console.log('Error in forex rates component. Message parameters:', snackBarParams)}
```

Reply:
I wanted it to remain so that the linter wouldn’t get angry that I hadn’t used snackbarParams yet.